### DISCUSSION TITLE
SSL certificate from environment

### DISCUSSION THREAD

**Discussion by jimmyspets** [aio-libs/aiohttp#9904] [Q&A] [answered]
I have the following code running for testing:

`ssl_context = ssl.create_default_context(
    cafile="./path/to/ca_certificate.cer"
)
`
`ssl_context.load_cert_chain(
    certfile="./path/to/certificate.cer", keyfile="./path/to/private_key.key"
)`

For production, I would like to store the CA certificate, client certificate, and private key in environment variables instead of using files.

For the CA certificate, I can decode it from an environment variable like this:
`ssl_context_env = os.getenv("SSL_CONTEXT")`
`ssl_context_str = base64.b64decode(ssl_context_env).decode("utf-8")`
`ssl_context = ssl.create_default_context(cadata=ssl_context_str)`

### Question
Is there an equivalent approach for handling the certfile and keyfile without saving them as temporary files? I would like to load the client certificate and private key directly from memory.

**Comment by asvetlov** [comment, marked as answer]
SSLContext doesn't provide an API for loading certs from memory yet.
As a workaround, you can get a certificate from enviar and put it in a temporary file for later loading.

**Reply by jimmyspets** [reply]
That's what I am doing right now, but for security reasons I would prefer not to.
